### ISSUE TITLE
Slow image loading and filter performance after 'big_images' merge, can cause oculante to crash

### ISSUE THREAD

**Issue by Stoppedpuma** [woelper/oculante#473]

**Comment by Stoppedpuma** [comment]
Error on my end, I was building in dev profile instead of release.

**Comment by woelper** [comment]
Not totally an error at your end, I think the opt-level decreased after merging, leading to slower performance. I'll fix that!

**Comment by Stoppedpuma** [comment]
@woelper I can confirm that something is leading to slower performance. I just tested with 2 week old branch and the loading performance is still fast with `cargo run` compared to taking several seconds now (as reported in #474). The blur is also faster on that branch. The branch for these tests is the `bundle_libheif_osx` branch.

**Comment by woelper** [comment]
I think at this point we need to profile. I also had the impression that something got slower, but I was blaming the opt-level change. I would encourage you to use `opt-level = 3` in Cargo.toml under `[profile.dev]` on master and look at the loading times in the terminal, then compare that with a branch that you believe to be fast. I think both the operator stack and file loading timings are displayed.

**Comment by Stoppedpuma** [comment]
Upon further testing, this seems to be an issue only with `cargo run`, builds aren't affected. Additionally, I'm not able to reproduce the performance I was getting before where the same 16K image would take nearly 10 seconds to open, nor the unusable blur performance on even images that are 256x256. I'm going to say that for now, this was an issue on my system, feel free to close this if you don't want to run your own benchmarks or aren't happy with current results and want to optimise.

Testing with `cargo run --no-default-features --features notan/shaderc`

`heif opt 1`: Received image in 1.959361349s

`heif opt 3`: Received image in 989.39947ms

`master opt 1`: Received image in 2.010625726s

`master opt 3`: Received image in 977.629584ms

Builds load the same 16K image in ~200ms on both master and heif branches.

**Comment by Stoppedpuma** [comment]
The performance I originally reported I can't reproduce any more, I have no clue how it was that bad during that test session, but it's now constantly around the numbers I posted. As of now, the conclusion I've made is that it's likely an issue with `cargo run` as builds are a lot faster, even if debug has the same flags as the release profiles.

I've only kept this issue open as woelper said "I also had the impression that something got slower, but I was blaming the opt-level change" which doesn't seem to be the case here. I'm going to close this now though as the issue doesn't seem to affect builds no matter what I try.
